virtual-desktop/src/app/start-url-manager/start-url-manager.service.ts

```diff
@@ -0,0 +1,141 @@
+    const allArgs = this.getAllApp2AppArgsFromURL();
+    if (!allArgs) {
+      return true;
+    }
+    for (const args of allArgs) {
+      if (this.validateApp2AppArgs(args)) {
+        this.invokeApp2App(args);
+      }
+    }
+    return true;
+  }
+
+  private validateApp2AppArgs(args: App2AppArgs): boolean {
+    const { dispatcher, pluginManager } = ZoweZLUX;
+    const plugin = pluginManager.getPlugin(args.pluginId);
+    if (!plugin) {
+      // this.logger.warn('Plugin not found for identifier %s specified in app2app query URL parameter', args.pluginId);
+      this.logger.warn('ZWED5193W', args.pluginId);
+      return false;
+    }
+    if (args.actionType !== 'launch' && args.actionType !== 'message') {
+      // this.logger.warn('Action Type(%s) specified  in app2app query URL parameter is invalid', args.actionType);
+      this.logger.warn('ZWED5194W', args.actionType);
+      return false;
+    }
+    if (args.actionMode !== 'create' && args.actionMode !== 'system') {
+      // this.logger.warn('Action Mode(%s) specified  in app2app query URL parameter is invalid', args.actionMode);
+      this.logger.warn('ZWED5195W', args.actionMode);
+      return false;
+    }
+    try {
+      JSON.parse(args.contextData);
+    } catch (e) {
+      // this.logger.warn('Context Data(%s) specified in app2app query URL parameter is not valid JSON', args.contextData);
+      this.logger.warn('ZWED5196W', args.contextData);
+      return false;
+    }
+    if (args.formatter !== 'data') {
+      const abstractAction = dispatcher.getAbstractActionById(args.formatter);
+      if (!abstractAction) {
+        // this.logger.warn('Unable to find registered action for identifier %s specified in app2app query URL parameter', args.formatter);
+        this.logger.warn('ZWED5197W', args.formatter);
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private invokeApp2App(args: App2AppArgs): void {
+    const { dispatcher } = ZoweZLUX;
+    const ActionType = dispatcher.constants.ActionType;
+    const ActionTargetMode = dispatcher.constants.ActionTargetMode;
+    const targetPluginId = args.pluginId;
+    const type = args.actionType === 'launch' ? ActionType.Launch : ActionType.Message;
+    const mode = args.actionMode === 'create' ? ActionTargetMode.PluginCreate : ActionTargetMode.System;
+    let contextData: any = JSON.parse(args.contextData);
+    let action: ZLUX.Action;
+    let argumentData: any;
+    if (args.formatter === 'data') {
+      const actionTitle = 'Launch app from URL';
+      const actionId = 'org.zowe.zlux.url.launch';
+      const argumentFormatter = { data: { op: 'deref', source: 'event', path: ['data'] } };
+      action = dispatcher.makeAction(actionId, actionTitle, mode, type, args.pluginId, argumentFormatter);
+      argumentData = { data: contextData };
+    } else {
+      const abstractAction = dispatcher.getAbstractActionById(args.formatter);
+      action = <ZLUX.Action>{
+        ...abstractAction,
```

Comment:
I don't remember the specifics, but old versions of Node (maybe 6) made it such that the spread operator doesn't work & wouldn't get compiled correctly into the JS, instead Object.assign was used

Don't remember if this is still an issue, it might have been an issue on Z for me, it's been nearly a year...

https://github.com/zowe/zlux-server-framework/commit/5abf39b0dd790b15cf1bf0ba6f533e30e1187792

Reply:
This is zlux-app-manager, where in multiple places currently ... is being used so it was probably only an issue for server core